flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/RemoteChannelStateChecker.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license agreements. See the NOTICE
+ * file distributed with this work for additional information regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package org.apache.flink.runtime.io.network.partition.consumer;
+
+import org.apache.flink.runtime.execution.ExecutionState;
+import org.apache.flink.runtime.io.network.partition.PartitionStateChecker;
+import org.apache.flink.runtime.io.network.partition.PartitionStateChecker.CheckResult;
+import org.apache.flink.runtime.io.network.partition.ResultPartitionID;
+import org.apache.flink.runtime.jobmanager.PartitionProducerDisposedException;
+import org.apache.flink.types.Either;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Handles the result of {@link PartitionStateChecker}.
+ *
+ * <p>The method {@code isProducerConsumerReadyOrAbortConsumption} determines whether the partition producer is
+ * in a producing state and consumer is ready for consumption.
+ */
+public class RemoteChannelStateChecker {
+	private static final Logger LOG = LoggerFactory.getLogger(RemoteChannelStateChecker.class);
+
+	private final ResultPartitionID resultPartitionId;
+
+	private final String taskNameWithSubtask;
+
+	public RemoteChannelStateChecker(ResultPartitionID resultPartitionId, String taskNameWithSubtask) {
+		this.resultPartitionId = resultPartitionId;
+		this.taskNameWithSubtask = taskNameWithSubtask;
+	}
+
+	public boolean isProducerConsumerReadyOrAbortConsumption(CheckResult checkResult) {
```

Comment:
Not super important and out of scope for this PR, but this method either triggers some action (fail or cancel) or returns a decision (trigger new partition check or not). I think it would be more symmetric if this class would not trigger any action but only return a decision what to do:
```
enum Action {
  FAIL(Throwable cause),
  CANCEL(String msg),
  TRIGGER_PARTITION_CHECK,
  NOOP
}
```
Then the caller would be responsible for making the action. That way this class would only need access to `checkResult.getProducerExecutionState()` and not `checkResult` itself.

Reply:
True, I would suggest we consider this refactoring as another issue

Reply:
https://issues.apache.org/jira/browse/FLINK-13638